### PR TITLE
Bluetooth: GATT: Add BT_GATT_CCC_MANAGED

### PR DESCRIPTION
**Pull request by Vudentz**
This adds another helper macro called BT_GATT_CCC_MANAGED which can be
used to set 2 new callbacks:

 - cfg_write: Which application can set if it wants to manage writes to
 CCC configuration.
 - cfg_match: Which application can set if it wants to manage matching
 CCC configuration when notifying/indicating.

BT_GATT_CCC_MANAGED retains the ability of saving peer configuration
on storage making it useful for clients which are only interrested in
managing the CCC configuration but not the storage itself.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

### CODE REVIEW

**Comment by Vudentz** [comment]
@pdunaj Any feedback on this?

**Comment by jhedberg** [comment]
@Vudentz thanks for updating Mesh code, but I think you missed the second GATT service in proxy.c. It has one service for provisioning and another for the GATT proxy, and both use the custom CCC tracking.

**Review by jhedberg — APPROVED** [review]
Fine with me - would be good if you could test that the mesh code still works though, e.g. try to provision mesh_shell using nRF Mesh (required shell commands: "mesh init" and "mesh pb-gatt enable", after which the device should be discoverable from nRF Mesh)

**Review by pdunaj — APPROVED** [review]
LGTM
left two small comments if you bother to fix code

File: subsys/bluetooth/host/gatt.c [hunk 1]
```diff
@@ -958,25 +958,38 @@ struct bt_gatt_attr *bt_gatt_attr_next(const struct bt_gatt_attr *attr)
 	return next;
 }
 
+static struct bt_gatt_ccc_cfg *find_ccc_cfg(struct bt_conn *conn,
+					    struct _bt_gatt_ccc *ccc)
+{
+	int i;
```
[the remark this answers is not in the document]
Comment (Vudentz) [reply subsys/bluetooth/host/gatt.c]: Will change it.

File: subsys/bluetooth/host/gatt.c [hunk 2]
```diff
@@ -958,25 +958,38 @@ struct bt_gatt_attr *bt_gatt_attr_next(const struct bt_gatt_attr *attr)
 	return next;
 }
 
+static struct bt_gatt_ccc_cfg *find_ccc_cfg(struct bt_conn *conn,
```
[the remark this answers is not in the document]
Comment (Vudentz) [reply subsys/bluetooth/host/gatt.c]: Not sure if this is worth it since it is a internal API

**Comment by Vudentz** [comment]
@jhedberg any other comments to resolve?

**Comment by jhedberg** [comment]
> @jhedberg any other comments to resolve?

@Vudentz Not from me at least. Now the question is whether this is something acceptable for merging for 1.14, since it's not a bug fix and not a documentation update.

**Pull request merged**